libraries/datahandler/src/main/thrift/vmcomponents.thrift

```diff
@@ -0,0 +1,150 @@
+}
+
+struct VMAction{
+    // General information
+    1: optional string id,
+    2: optional string revision,
+    3: optional string type = "vmaction",
+    4: optional string lastUpdateDate,
+
+    // Additional information
+    10: required string vmid,   // 5
+    11: optional string text,   // "text": "Install New Package"
+}
+
+struct VMComponent{
+    // Basic information
+    1: optional string id,
+    2: optional string revision,
+    3: optional string type = "vmcomponent",
+    5: required string receivedDate,
+    6: optional string lastUpdateDate,
+
+    // Additional information
+    10: required string vmid,       // 70
+    11: optional string vendor,     // "vendor":         "Apache Software Foundation",
+    12: optional string name,       // "component_name": "Tomcat",
+    13: optional string version,    // "version":        "3.2.1",
+    14: optional string url         // "url":            "http://tomcat.apache.org/",
+    15: optional string securityUrl,// "security_url":   null,
+    16: optional bool eolReached, // "eol_reached":    true,
+    17: optional string cpe,        // "cpe_name":       "cpe:/a:apache:tomcat:3.2.1",
+
+    // additional information about patches
+    20: optional set<VMMinPatchLevel> minPatchLevels, // "minimum_patch_levels": {"priority_1": null,"priority_2": "6.0.36","priority_3": null }
+}
+
+struct VMMinPatchLevel{
+    // Basic information
+    1: optional string id,
+    2: optional string revision,
+    3: optional string type = "vmminpatchlevel",
+
+    // Additional information
+    10: required string priority, // vmid of vmpriority
+    11: optional string version,
+}
+
+struct VMMatch{
+    // Basic information
+    1: optional string id,
+    2: optional string revision,
+    3: optional string type = "vmmatch",
+
+    // vm component information
+    11: required string vmComponentId;
+    12: optional string vmComponentCpe;
+    13: optional string vmComponentName;
+    14: optional string vmComponentVendor;
+    15: optional string vmComponentVersion;
+    16: optional string vmComponentVmid;
+
+    // sw360 release information
+    21: required string releaseId;
+    22: optional string releaseCpe;
+    23: optional string componentName;
+    24: optional string vendorName;
+    25: optional string releaseVersion;
+    26: optional string releaseSvmId;
+
+    // matching information
+    31: required set<VMMatchType> matchTypes,
+        32: optional string matchTypesUI;
+    33: required VMMatchState state,
+}
+
+service VMComponentService {
+    // General information
+    list<VMProcessReporting> getAllProcesses(1: User user);
+    list<VMMatch> getAllMatches(1: User user);
+
+    // Trigger for vulnerability monitoring
+    RequestSummary synchronizeComponents(/*1: User user*/);
+    RequestSummary triggerReverseMatch(/*1: User user*/);
+//    RequestSummary scheduleSync(1: User user);
+//    RequestSummary unscheduleSync(1: User user);
```

Comment:
remove the commented lines (`146` and `147`). also the commented method parameters in line `144` and `145`.